mersenne-31/src/extension.rs

```diff
@@ -0,0 +1,80 @@
+use p3_field::extension::cubic::CubicOef;
+use p3_field::extension::quadratic::QuadraticOef;
+use p3_field::extension::OptimallyExtendable;
+
+use crate::{Mersenne31, Mersenne31Complex};
+
+impl OptimallyExtendable<2> for Mersenne31Complex<Mersenne31> {
+    type Extension = QuadraticOef<Self>;
+
+    // Verifiable in Sage with
+    // `p = 2**31 - 1  # Mersenne31`
+    // `F = GF(p)  # The base field GF(p)`
+    // `R = F['x']  # The polynomial ring over F`
+    // `K = F.extension(R.gen()**2 + 1, 'i')  # The extension field of complex`
+    // `R2 = K['y']`
+    // `f2 = R2.gen()**2-K('i')-2   # polynomial y^2-i-2`
+    // `f.is_irreducible()`
+    const W: Self = Self::new(Mersenne31::new(2), Mersenne31::new(1));
+
+    // DTH_ROOT = W^((p - 1)/2)
+    const DTH_ROOT: Self = Self::new(Mersenne31::new(21189756), Mersenne31::new(42379512));
+
+    // Verifiable in Sage with
+    //  `g = K2('j')+6`
+    //  `for f,i in list(factor(p**4-1)):`
+    //  `  print("j to ",Integer(f)," : ",g**Integer((p^4-1)/f))`
```

Comment:
```suggestion
    // Verifiable in Sage with
    // ```sage
    // K2.<j> = K.extension(f2)
    //  g = j + 6
    // for f in factor(p^4 - 1):
    //   assert (j+6)^((p^4-1) // f) != 1 
    // ```
```